build_tools/benchmarks/export_benchmark_config.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Exports JSON config for benchmarking.
+
+The exported JSON is a list of object:
+[
+  <target device name>: {
+    host_environment: HostEnvironment,
+    module_dir_paths: [<paths of dependent module directories>],
+    run_configs: [E2EModelRunConfig]
+  },
+  ...
+]
+"""
+
+import sys
+import pathlib
+
+# Add build_tools python dir to the search path.
+sys.path.insert(0, str(pathlib.Path(__file__).parent.with_name("python")))
+
+from typing import List
+import argparse
+import collections
+import dataclasses
+import itertools
+import json
+
+from benchmark_suites.iree import benchmark_collections
+from e2e_test_framework import serialization
+from e2e_test_artifacts import iree_artifacts
+
+
+def sort_and_dedup_paths(
+    values: List[pathlib.PurePath]) -> List[pathlib.PurePath]:
+  values.sort()
+  return list(value for value, _ in itertools.groupby(values))
```

Comment:
I think generally it's better to not modify your input unless that's the purpose of the function or it's performance critical. I don't think either is the case here, so I'd recommend `sorted(values)` instead.

Additionally I think using a simpler uniquer is perhaps easier to grok (and again, not sure this is performance critical. What about this (combining with the suggestion below)

```suggestion
    values: Iterable[pathlib.PurePath]) -> List[pathlib.PurePath]:
  return sorted(set(values))
```

Although actually at that point, maybe the function isn't even necessary!

Reply:
The `sorted(set(values))` looks good to me. Thx!

I do need to write it into two lines; otherwise yapf will format it into a weird way...